### ISSUE TITLE
ridk exec pacman failed in Windows 10

### ISSUE THREAD

**Issue by ShisiJu** [puma/puma#2618]
- puma Version 4.3.7 
- ruby 2.6.5-p114 

I want to debug puma, and I see the `CONTRIBUTING.md`

Windows (Ruby 2.5 and later):

```sh
ridk exec pacman -S mingw-w64-x86_64-openssl mingw-w64-x86_64-ragel
```

I run the script , but failed.  The erros shows like the following image. It means that it can not find the files, so installing failed.

Do you know why?  And how can I install regal in Windows 10 ?   Thanks in advance!

![image](https://user-images.githubusercontent.com/39001403/116835681-1e161400-abf6-11eb-83fc-fab62e39a8fc.png)

**Comment by MSP-Greg** [comment]
@ShisiJu

First, this can get messy.  MSYS2 (the build tools used for Windows Ruby and gems) is an independent open source project.   There have been a lot of changes in the past several months.  The changes have involved both download locations and package signing gpg keys.  

Hence, I always recommend that people use the latest 'patch' version of Ruby, which in your case would be 2.6.7.

The simplest way to update MSYS2 is to keep performing the following until you see something in the console similar to 'there is nothing to do'.  It may close the terminal window a few times:
```
ridk exec pacman -Syu
```

After that, run the command you listed, which installs openssl and ragel.

> I want to debug puma

Not sure what you mean by debug, as in a c debugger, or a 'Ruby level' debugger.  Puma is currently at v5.2.2...

**Comment by ShisiJu** [comment]
Thanks for you reply.

>Not sure what you mean by debug, as in a c debugger, or a 'Ruby level' debugger. Puma is currently at v5.2.2...

I am just curious about how puma handle the HTTP request.  So I want to set a debugger in `Ruby`.   

**My goal is that run the puma code, then I can inspect.**  I saw the [The Rack specification](https://github.com/rack/rack/blob/master/SPEC.rdoc), and I want to know more detail.

---

Maybe I do not need to install regal?

**Comment by MSP-Greg** [comment]
Unfortunately, to either install the gem or compile from the repo, you should have the OpenSSL package installed.  If your system can install that, then ragel can be installed with the same command, so ragel really isn't any more work.

The http parser requires compiling, regardless of whether your system has OpenSSL or Ragel.  You should be able to install the gem (or build the repo) without OpenSSL if you set `$env:DISABLE_SSL = 'true'` (PowerShell) or `SET DISABLE_SSL=true` (cmd).

I'd recommend getting your MSYS2 up to date, then you can compile/install Puma with OpenSSL support...

**Comment by ShisiJu** [comment]
Got it!  Thanks for your patience!
I will update my MSYS2, then try again.

**Comment by MSP-Greg** [comment]
@ShisiJu

Above I said "that people use the latest 'patch' version of Ruby" and "this can get messy".  I thought a better explanation might be in order.

Ruby patch versions contain bug fixes for previous versions.  So, using 2.6.7 is probably better than using an older version of 2.6.

With Windows MinGW, there is another reason, and that is build tools compatibility.  Package managers vary as to whether they allow users to select what version of a package they install.  Pacman doesn't normally allow installing older packages.  One can download older packages with other tools and install them.

So, using Ruby 2.6 as an example, when 2.6.0 was released, it was compiled with gcc 8.2.1.  Releases after that used gcc 8.3.0, 9.2.0, and 9.3.0.  The most recent release, 2.6.7, used gcc 10.2.0.

MSYS2 just released gcc 10.3.0.

Obviously, the organization behind gcc does their best to avoid breaking changes between versions, but that isn't always possible.

Hence, using older Ruby patch versions with an updated MSYS2 build system may cause issues.
